src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const NEW_STATE = { ...state };
+  const STATES = [];
+
+  for (const ACTION of actions) {
+    switch (ACTION.type) {
+      case 'addProperties':
+        Object.assign(NEW_STATE, ACTION.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const ELEM of ACTION.keysToRemove) {
+          delete NEW_STATE[ELEM];
+        }
+        break;
+
+      case 'clear':
+        for (const KEY in NEW_STATE) {
```

Comment:
The variable 'KEY' should be in camelCase to follow JavaScript naming conventions, such as 'key'.